tools/autograd/gen_variable_type.py

```diff
@@ -73,9 +194,90 @@ def write(dirname, name, template, env):
 
 def load_derivatives(path):
     with open(path, 'r') as f:
-        # TODO: load and parse derivatives.yaml
-        d = yaml.load(f, Loader=Loader)
-        return [] if d is None else d
+        definitions = yaml.load(f, Loader=Loader)
+
+    name_regex = r'(^|\W){}($|\W)'
+
+    def split_name_params(prototype):
+        name, params = re.match('(\w+)\((.*)\)', prototype).groups()
+        return name, params.split(', ')
+
+    def get_signature(option):
+        arguments = option['python_arguments']
+        arg_types = [arg['type'] for arg in arguments]
+        if option['aten'] is not None:
+            call_args = split_name_params(option['aten'])[1]
+            arg_indices = {arg['name']: i for i, arg in enumerate(arguments)}
+
+            def get_type(arg_name):
+                if arg_name in arg_indices:
+                    return arg_types[arg_indices[arg_name]]
+                return 'Scalar'
+
+            arg_types = [get_type(arg_name) for arg_name in call_args]
+        return '{}({})'.format(option['name'], ', '.join(arg_types))
```

Comment:
It feels weird that function name is part of the signature 😕 I've been pretty sure we haven't been handling similar signatures correctly until I realized that